src/server/scripts/Kalimdor/VortexPinnacle/boss_asaad.cpp

```diff
@@ -0,0 +1,775 @@
+
+                _JustDied();
+            }
+
+            void UpdateAI(uint32 diff) override
+            {
+                if (!UpdateVictim())
+                    return;
+
+                events.Update(diff);
+
+                if (me->HasUnitState(UNIT_STATE_CASTING))
+                    return;
+
+                while (uint32 eventId = events.ExecuteEvent())
+                {
+                    switch (eventId)
+                    {
+                        case EVENT_STATIC_CLING:
+                            DoCast(me, SPELL_STATIC_CLING);
+                            events.ScheduleEvent(EVENT_STATIC_CLING, 16000);
+                            break;
+                        case EVENT_SUMMON_SKYFALL_STAR:
+                            DoCast(me, SPELL_SUMMON_SKYFALL_STAR);
+                            events.ScheduleEvent(EVENT_CHAIN_LIGHTNING, 14500);
+                            break;
+                        case EVENT_CHAIN_LIGHTNING:
+                            DoCast(me, SPELL_CHAIN_LIGHTNING);
+                            events.ScheduleEvent(EVENT_CHAIN_LIGHTNING, 14500);
+                            break;
+                        case EVENT_SOTS:
+                            Talk(SAY_SOTS_EMOTE);
+                            events.Reset();
+                            events.ScheduleEvent(EVENT_SOTS_START, 200);
+                            events.ScheduleEvent(EVENT_SOTS, 45800);
+                            break;
+                        case EVENT_SOTS_START:
+                            me->SetReactState(REACT_PASSIVE);
+                            me->AttackStop();
+                            Talk(SAY_SOTS);
+                            DoCast(me, SPELL_SOTS_TARGETING);
+                            break;
+                        case EVENT_SOTS_END:
+                            ResetStormTargets();
+                            me->SetDisableGravity(false);
+                            events.ScheduleEvent(EVENT_ATTACK, 2000);
+                            break;
+                        case EVENT_ATTACK:
+                            me->SetReactState(REACT_AGGRESSIVE);
+                            events.ScheduleEvent(EVENT_CHAIN_LIGHTNING, 100);
+                            events.ScheduleEvent(EVENT_STATIC_CLING, 2000);
+                            events.ScheduleEvent(EVENT_SUMMON_SKYFALL_STAR, 3500);
+                            break;
+                        default:
+                            break;
+                    }
+                }
+
+                DoMeleeAttackIfReady();
+            }
+
+            Creature* GetSelectedStormTarget(uint32 point)
+            {
+                if (point == POINT_STORM_A)
+                    return stormTargetA;
+                if (point == POINT_STORM_B)
+                    return stormTargetB;
+                if (point == POINT_STORM_C)
+                    return stormTargetC;
+                return NULL;
+            }
```

Comment:
deal with uint64 UnitAI::GetGUID(int32 /*id*/ = 0) const { return 0; } to get rid of these ai casts in other scripts